src/effects/effectslot.cpp

```diff
@@ -253,13 +259,15 @@ void EffectSlot::setEnabled(bool enabled) {
     m_pControlEnabled->set(enabled);
 }
 
-EffectKnobParameterSlotPointer EffectSlot::getEffectKnobParameterSlot(unsigned int slotNumber) {
+EffectKnobParameterSlotPointer EffectSlot::getEffectKnobParameterSlot(unsigned int position) {
     //qDebug() << debugString() << "getEffectKnobParameterSlot" << slotNumber;
+    int slotNumber = m_knobParameterPositionToSlot.value(position, -1);
```

Comment:
Revert this. When hidden parameters are not loaded in parameter slots, this would require other classes to know which parameters are hidden to get the parameter slot they are searching for. 

Eventually this function should be removed entirely because other classes should not have direct access to parameter slots. All access to the effects system from outside the effects system should be mediated by EffectsManager to decouple the GUI from implementation details. Please add a comment noting this.

The same applies for getEffectButtonParameterSlot.